eo-parser/src/main/java/org/eolang/parser/XeListener.java

```diff
@@ -180,7 +186,15 @@ public void enterObject(final ProgramParser.ObjectContext ctx) {
             if (application.suffix() != null) {
                 application = application.application();
             }
-            this.check.accept(application.getText());
+            final String text = application.getText();
+            if (this.check.test(text)) {
+                this.errors.xpath("/program/errors")
```

Comment:
@volodya-lombrozo a more elegant and shorter solution:

`this.dirs.push().xpath("/program/errors").add("error")....pop()`

Reply:
@yegor256 Works like a charm. Thank you. Fixed.